packages/react/src/tooltip/popup/TooltipPopup.tsx

```diff
@@ -27,12 +29,28 @@ export const TooltipPopup = React.forwardRef(function TooltipPopup(
   componentProps: TooltipPopup.Props,
   forwardedRef: React.ForwardedRef<HTMLDivElement>,
 ) {
-  const { className, render, ...elementProps } = componentProps;
+  const { className, render, id: idProp, ...elementProps } = componentProps;
 
-  const { open, instantType, transitionStatus, popupProps, popupRef, onOpenChangeComplete } =
-    useTooltipRootContext();
+  const id = useBaseUiId(idProp);
+
+  const {
+    open,
+    instantType,
+    transitionStatus,
+    popupProps,
+    popupRef,
+    onOpenChangeComplete,
+    setPopupId,
+  } = useTooltipRootContext();
   const { side, align } = useTooltipPositionerContext();
 
+  useIsoLayoutEffect(() => {
+    setPopupId(id);
+    return () => {
+      setPopupId(undefined);
+    };
+  }, [setPopupId, id]);
```

Comment:
Perf note: this pattern forces a re-render right after the popup mounts/renders for the first time.